app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,277 @@
+        flex: 1,
+    },
+    searchBar: {
+        marginLeft: 12,
+        marginRight: Platform.select({ios: 4, default: 12}),
+        marginVertical: 12,
+    },
+});
+
+const messages = defineMessages({
+    button_manage: {
+        id: t('mobile.manage_members.manage'),
+        defaultMessage: 'Manage',
+    },
+    button_done: {
+        id: t('mobile.manage_members.done'),
+        defaultMessage: 'Done',
+    },
+});
+
+const MANAGE_BUTTON = 'manage-button';
+const EMPTY: UserProfile[] = [];
+const EMPTY_MEMBERS: ChannelMembership[] = [];
+const EMPTY_IDS = {};
+const {USER_PROFILE} = Screens;
+const CLOSE_BUTTON_ID = 'close-user-profile';
+
+export default function ManageChannelMembers({
+    canManageAndRemoveMembers,
+    channelId,
+    componentId,
+    currentTeamId,
+    currentUserId,
+    teammateNameDisplay,
+    tutorialWatched,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const {formatMessage} = useIntl();
+
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const mounted = useRef(false);
+
+    const [isManageMode, setIsManageMode] = useState(false);
+    const [profiles, setProfiles] = useState<UserProfile[]>(EMPTY);
+    const [channelMembers, setChannelMembers] = useState<ChannelMembership[]>(EMPTY_MEMBERS);
+    const [searchResults, setSearchResults] = useState<UserProfile[]>(EMPTY);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+
+    const loadedProfiles = (users: UserProfile[], members: ChannelMembership[]) => {
+        if (mounted.current) {
+            setLoading(false);
+            setProfiles(users);
+            setChannelMembers(members);
+        }
+    };
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults(EMPTY);
+    }, []);
+
+    const getProfiles = useCallback(debounce(async () => {
+        const hasTerm = Boolean(term);
+        if (!loading && !hasTerm && mounted.current) {
+            setLoading(true);
+            const options = {sort: 'admin', active: true};
+            const {users, members} = await fetchChannelMemberships(serverUrl, channelId, options, true);
+            if (users.length) {
+                loadedProfiles(users, members);
+            }
+        }
+    }, 100), [channelId, loading, serverUrl, term]);
+
+    const handleSelectProfile = useCallback(async (profile: UserProfile) => {
+        const title = formatMessage({id: 'mobile.routes.user_profile', defaultMessage: 'Profile'});
+        const props = {
+            channelId,
+            closeButtonId: CLOSE_BUTTON_ID,
+            location: USER_PROFILE,
+            manageMode: isManageMode,
+            userId: profile.id,
+            canManageAndRemoveMembers,
+        };
+
+        Keyboard.dismiss();
+        openAsBottomSheet({screen: USER_PROFILE, title, theme, closeButtonId: CLOSE_BUTTON_ID, props});
+    }, [canManageAndRemoveMembers, isManageMode]);
```

Comment:
Theoretically, this depends on the channelId and the profile.id. Not sure if left out intentionally or not.

Reply:
channelId should be a dependency, but profile.id is a prop to this function.